fluent/syntax/parser.py

```diff
@@ -27,7 +51,12 @@ def parse(self, source):
 
             ps.skip_ws_lines()
 
-        return ast.Resource(entries, comment)
+        res = ast.Resource(entries, comment)
+
+        if self.with_spans:
+            res.add_span(0, ps.get_index())
```

Comment:
`parse` is part of the public API and doesn't take `ps` (the `FTLParserStream` instance) as argument. In fact, it creates it for the purpose of parsing.  `ps` is required to get span positions.